ee/hogai/graph/graph.py

```diff
@@ -292,13 +275,59 @@ def add_sql_generator(self, next_node: AssistantNodeName = AssistantNodeName.QUE
 
         return self
 
-    def add_query_executor(self, next_node: AssistantNodeName = AssistantNodeName.ROOT):
+    def add_query_executor(self, next_node: AssistantNodeName = AssistantNodeName.END):
         builder = self._graph
         query_executor_node = QueryExecutorNode(self._team)
         builder.add_node(AssistantNodeName.QUERY_EXECUTOR, query_executor_node)
         builder.add_edge(AssistantNodeName.QUERY_EXECUTOR, next_node)
         return self
 
+    def compile_full_graph(self):
+        return (
+            self.add_rag_context()
+            .add_trends_planner()
+            .add_trends_generator()
+            .add_funnel_planner()
+            .add_funnel_generator()
+            .add_retention_planner()
+            .add_retention_generator()
+            .add_sql_planner()
+            .add_sql_generator()
+            .add_query_executor()
+            .compile()
+        )
+
+
+class AssistantGraph(_AssistantGraph):
+    def add_root(
+        self,
+        path_map: Optional[dict[Hashable, AssistantNodeName]] = None,
+    ):
+        builder = self._graph
+        path_map = path_map or {
+            "product_analytics": AssistantNodeName.PRODUCT_ANALYTICS_SUBGRAPH,
+            "search_documentation": AssistantNodeName.INKEEP_DOCS,
+            "root": AssistantNodeName.ROOT,
+            "end": AssistantNodeName.END,
+        }
+        root_node = RootNode(self._team)
+        builder.add_node(AssistantNodeName.ROOT, root_node)
+        root_node_tools = RootNodeTools(self._team)
+        builder.add_node(AssistantNodeName.ROOT_TOOLS, root_node_tools)
+        builder.add_edge(AssistantNodeName.ROOT, AssistantNodeName.ROOT_TOOLS)
+        builder.add_conditional_edges(
+            AssistantNodeName.ROOT_TOOLS, root_node_tools.router, path_map=cast(dict[Hashable, str], path_map)
+        )
+        return self
+
+    def add_product_analytics_subgraph(self, next_node: AssistantNodeName = AssistantNodeName.ROOT):
+        builder = self._graph
+        product_analytics_subgraph = InsightsAssistantGraph(self._team)
```

Comment:
Naming would be a bit clearer with extra consistency: `product_analytics_subgraph` vs. `InsightsAssistantGraph`